### PR TITLE
X11: Use window shape to fix input for invisible windows

### PR DESCRIPTION
**Pull request by leolost2605**
The first commit just reverts #2122 

Rebase merge

Fixes the same issue as #2122 and is already prepared to be used as intended in #2134. Since it doesn't move anything we don't run into issues of having to solve conflicts between e.g. window positioner and the hidden window, as well as not having to have an additional class handle coordinates etc. which is IMHO quite a bit cleaner.

 Closes #2134

### CODE REVIEW

File: lib/Utils.vala [hunk 1]
```diff
@@ -400,5 +400,50 @@ namespace Gala {
 
             return texture;
         }
+
+        private static HashTable<Meta.Window, X.XserverRegion?> regions = new HashTable<Meta.Window, X.XserverRegion?> (null, null);
+
+        public static void x11_set_window_pass_through (Meta.Display display, Meta.Window window) {
+            unowned var x11_display = display.get_x11_display ();
```
Comment (lenemter) [inline lib/Utils.vala:407]:
```suggestion
        public static void x11_set_window_pass_through (Meta.Window window) {
            unowned var x11_display = window.display.get_x11_display ();
```

File: lib/Utils.vala [hunk 2]
```diff
@@ -400,5 +400,50 @@ namespace Gala {
...
+
+            var region = X.Fixes.create_region (xdisplay, {rect});
+
+            X.Fixes.set_window_shape_region (xdisplay, x_window, 2, 0, 0, region);
+
+            X.Fixes.destroy_region (xdisplay, region);
+        }
+
+        public static void x11_unset_window_pass_through (Meta.Display display, Meta.Window window) {
+            unowned var x11_display = display.get_x11_display ();
```
Comment (lenemter) [inline lib/Utils.vala:428]:
```suggestion
        public static void x11_unset_window_pass_through (Meta.Window window) {
            unowned var x11_display = window.display.get_x11_display ();
```

**Review by lenemter — CHANGES_REQUESTED** [review]
Tried the same thing before https://github.com/elementary/gala/pull/2122 but it didn't work. Here it works flawlessly ¯\\\_(ツ)_/¯

File: lib/Utils.vala (hunk 1 above)
Comment (leolost2605) [reply lib/Utils.vala]: Ah I always forget about that :)

**Review by lenemter — APPROVED** [review]
Thanks for looking into this <3

**Pull request merged**